cmd/prometheus/main.go

```diff
@@ -443,7 +443,11 @@ func main() {
 				}
 				files = append(files, fs...)
 			}
-			return ruleManager.Update(time.Duration(cfg.GlobalConfig.EvaluationInterval), files)
+			return ruleManager.Update(
+				time.Duration(cfg.GlobalConfig.EvaluationInterval),
+				files,
+				cfg.GlobalConfig.ExternalLabels,
```

Comment:
I would have passed the entire `cfg.GlobalConfig` not just a subset. Like that in the future if you need to access new global configurations you wouldn't have to redo this tedious work.

Reply:
Yeah, I was thinking about that, too. Some of the `reloaders` do it that way, some not. It has pros and cons. If you pass down the whole `cfg.GlobalConfig`, you create a tight coupling of the rule manager to the config. On the other hand, the more parts of the config you are using, the more sense it makes.

I don't feel strongly and decided to stick with the way it has been done so far (which even includes the globbing logic). One might argue the one more thing to pass down flips the balance, but I don't think so.

@brian-brazil what do you think?

Reply:
I don't really mind either way, but given the code is reused by other projects it's probably best to keep the coupling minimal.

Reply:
Yeah, ideally each package defines its own config structs that are relevant to it and then the global config only ties those together. At least we were moving into that direction, even if not everything is doing that yet.